### ISSUE TITLE
Supporting documentation for Angular 2.x use / consider removing dependency

### ISSUE THREAD

**Issue by arimus** [mermaid-js/mermaid#1082]
**Is your feature request related to a problem? Please describe.**
Integration of mermaid into an Angular 2.x project may result in an error like:

```
ERROR Error: Uncaught (in promise): ReferenceError: global is not defined
ReferenceError: global is not defined
    at Object../node_modules/node-libs-browser/node_modules/buffer/index.js (index.js:43)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/safe-buffer/index.js (index.js:2)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/randombytes/browser.js (browser.js:15)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/crypto-browserify/index.js (index.js:3)
    at __webpack_require__ (bootstrap:84)
    at Object../node_modules/crypto-random-string/index.js (index.js:2)
    at __webpack_require__ (bootstrap:84)
    at resolvePromise (zone-evergreen.js:797)
    at resolvePromise (zone-evergreen.js:754)
    at zone-evergreen.js:858
    at ZoneDelegate.invokeTask (zone-evergreen.js:391)
    at Object.onInvokeTask (core.js:39679)
    at ZoneDelegate.invokeTask (zone-evergreen.js:390)
    at Zone.runTask (zone-evergreen.js:168)
    at drainMicroTaskQueue (zone-evergreen.js:559)
```

This can be resolved by polyfill'ng adding the following to polyfills.ts:

```
(window as any).global = window; // gets rid of the first error
global.Buffer = global.Buffer || require('buffer').Buffer; // which leads to buffer being required
(window as any).process = { // which leads to window.process.version needing to be parsed
  env: { DEBUG: undefined }, // https://github.com/nodejs/readable-stream/issues/313
  version: 'v0.9.'
};
(window as any).setImmediate = window.setTimeout; // which leads to setImmediate being required
```

Note that all of this is due to 'crypto-random-string' which relies on 'crypto' which pulls in a bunch of wedges intended to emulate node libs in the browser, some of which still have issues without the above polyfills.

**Describe the solution you'd like**
At a minimum, documentation for the polyfills above. Ideally, I'd remove or change out the crypt string library for a simpler dependency chain (which would also be smaller).

Thanks for the cool library!

**Comment by knsv** [comment]
It should be possible to replace crypto-random-string.

**Comment by tjrobinson** [comment]
We are facing a similar issue with Angular 8.

The error is:

```
ERROR in ./node_modules/crypto-random-string/index.js
Module not found: Error: Can't resolve 'crypto' in '.\node_modules\crypto-random-string'
```

We have worked around it for now by adding this to the `package.json`:

```
  "browser": {
    "crypto": false,
  }
```

Based on this suggestion for a similar problem in another library: https://github.com/iden3/snarkjs/issues/17#issuecomment-497218961

To explain the workaround:

`mermaid` has a dependency on `crypto-random-string` (which is only used for the git graphs) which includes this line: `const crypto = require('crypto');`

`crypto` is a Node JS built-in module: https://nodejs.org/api/crypto.html#crypto_crypto

This isn't available when running in a browser (as far as I can tell, I'm not an expert).

The `browser` section in `package.json` is "a hint to javascript bundlers or component tools when packaging modules for client side use". See: https://github.com/defunctzombie/package-browser-field-spec

In this particular case, it says that `crypto` should be ignored when requested: https://github.com/defunctzombie/package-browser-field-spec#ignore-a-module

This works for us but I expect the git graphs would fail to work.

**Comment by arimus** [comment]
So, in my case it actually fetches the crypto-browserify dep when it detects a browser environment. And clearly tries to do the same all the way down the chain, however these browser deps don’t seem to bring their own polyfills or move on when they don’t have access to a few things. Apparently, some of this is related to webpack version, which used to auto provide global, etc. Either way, it’s kludgy and a pain and would be awesome to just not deal with simply for getting a random string that I’m gonna guess does not need to be cryptographically secure ;)

**Comment by chris579** [comment]
I think you don't intended to close this issue.

**Comment by arimus** [comment]
@chris579 Indeed. Oops! Was on a temp phone when I commented...and must have clicked the wrong button. That's a first for me. My apologies!

**Comment by chris579** [comment]
Probably related to #1175. On hold until the other issue is resolved.

**Comment by jgreywolf** [comment]
As #1175 was closed as "wont fix", we should revisit this issue.  Was this only an issue on IE11?  Is this still something that needs to be handled?

**Comment by arimus** [comment]
This is not just an issue in IE. If I recall, crypto-random-string is a
nodejs module and does not work in any browser. I was testing in chrome
when I ran into this issue.

**Comment by jgreywolf** [comment]
@arimus is this still an issue after all of the work that has been done since 2021?

**Comment by jgreywolf** [comment]
Closing due to lack of response/activity
